packages/cli/src/index.js

```diff
@@ -31,69 +32,83 @@ import * as typeCheckCommand from './commands/type-check'
 import * as upgradeCommand from './commands/upgrade'
 import { getPaths } from './lib'
 
-/**
- * The current working directory can be set via:
- * 1. A `--cwd` option
- * 2. The `RWJS_CWD` env-var
- * 3. Found by traversing directories upwards for the first `redwood.toml`
- *
- * This middleware parses, validates, and sets current working directory
- * in the order above.
- */
-const getCwdMiddleware = (argv) => {
-  let configPath
+// # Setting the CWD
+//
+// The current working directory can be set via:
+//
+// 1. The `--cwd` option
+// 2. The `RWJS_CWD` env-var
+// 3. By traversing directories upwards for the first `redwood.toml`
+//
+// ## Examples
+//
+// ```
+// yarn rw info --cwd /path/to/project
+// RWJS_CWD=/path/to/project yarn rw info
+//
+// # In this case, `--cwd` wins out
+// RWJS_CWD=/path/to/project yarn rw info --cwd /path/to/other/project
+//
+// # Here `findup` traverses upwards.
+// cd api
+// yarn rw info
+// ```
+let { cwd } = parser(hideBin(process.argv))
+cwd ??= process.env.RWJS_CWD
+const redwoodTomlPath = findup('redwood.toml', { cwd: cwd ?? process.cwd() })
```

Comment:
Why don't we fall back to `process.cwd()` for `cwd`? I.e. why don't we do
`cwd ??= process.env.RWJS_CWD || process.cwd()`?

Reply:
Because if it wasn't specified by the `--cwd` flag (takes first priority) or the `RWJS_CWD` env var (takes second priority), we set it by "finding up" for `redwood.toml` later here:

https://github.com/redwoodjs/redwood/blob/b61a3f624ff502023218fa88e10f40e5a7fa5355/packages/cli/src/index.js#L76

Leaving `cwd` undefined was the easiest way to know that I still have to set it later, but happy to approach it from another way if it's hard to read, just what I came up with at the time

Reply:
@Tobbe let me try something related to https://github.com/redwoodjs/redwood/pull/6199. I'll probably rework this a bit either way

Reply:
@Tobbe that file's ready for another review, but no rush! Also added a test for it: https://github.com/redwoodjs/redwood/blob/51ffd8c0cceb0df19c0d5eb04f0185b9584058af/packages/cli/src/index.test.js.

It felt a little weird writing a test in src that uses execa to run the the transpiled version in dist... but I wasn't sure what else to do 😅

Reply:
I like all the comments you have in the file. Makes it really easy to understand what's going on.

I agree the testing strategy feels a little bit weird. If you have the tests running in watch mode and you make updates to the implementation, do they even pick up on the changes? I.e. is the code continuously built along with the tests? Either way, at least the tests are really clear and do demonstrate the expected behavior in a good way.